File: julia/bob/bob.jl
function bob(stimulus)
	stimulus = rstrip(stimulus)
    nonlower = true
	hasletter = false
    for c in collect(stimulus)
		hasletter = !hasletter ? isletter(c) : true
        nonlower = nonlower ? !isletter(c) || isuppercase(c) : false
    end
	allupper = hasletter && nonlower

    if isempty(stimulus)
        "Fine. Be that way!"
    elseif endswith(stimulus, '?')
        allupper ? "Calm down, I know what I'm doing!" : "Sure."
    elseif allupper
        "Whoa, chill out!"
    else
        "Whatever."
    end
end

File: csharp/bob/Bob.cs
using System.Linq;

public static class Bob
{
    public static string Response(string statement)
    {
        if (string.IsNullOrWhiteSpace(statement))
            return "Fine. Be that way!";

        statement = statement.TrimEnd();

        var hasLetter = false;
        var noLower = statement.All(c =>
        {
            if (!char.IsLetter(c))
                return true;

            hasLetter = true;
            return char.IsUpper(c);
        });
        var allUpper = hasLetter && noLower;

        return statement[^1] == '?' ? allUpper ? "Calm down, I know what I'm doing!" : "Sure." : allUpper ? "Whoa, chill out!" : "Whatever.";
    }
}